hackathon_site/dashboard/frontend/src/components/team/TeamSearchBar/TeamSearchBar.tsx

```diff
@@ -5,32 +5,52 @@ import InputAdornment from "@material-ui/core/InputAdornment";
 import IconButton from "@material-ui/core/IconButton";
 import CloseIcon from "@material-ui/icons/Close";
 import SearchIcon from "@material-ui/icons/Search";
-import React from "react";
+import { getTeamsWithSearchThunk } from "slices/event/teamAdminSlice";
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
 
-const TeamSearchBar = () => (
-    <form autoComplete="off" className={styles.searchBar}>
-        <Box display="flex" flexDirection="row">
-            <TextField
-                id="search-input"
-                name="search"
-                label="Search items"
-                variant="outlined"
-                type="text"
-                InputProps={{
-                    endAdornment: (
-                        <InputAdornment position="end">
-                            <IconButton>
-                                <CloseIcon />
-                            </IconButton>
-                        </InputAdornment>
-                    ),
-                }}
-            />
-            <IconButton color="primary" aria-label="Search">
-                <SearchIcon />
-            </IconButton>
-        </Box>
-    </form>
-);
+const TeamSearchBar = () => {
```

Comment:
Instead of setState and handleChange, could you implement Formik for the search bar?